packages/store/src/Bytes.sol

```diff
@@ -0,0 +1,705 @@
+      // so we add 32 to the pointer to get to the start of the data
+      // then we add the start offset to get to the start of the desired word
+      output := mload(add(input, add(0x20, offset)))
+    }
+  }
+
+  function toBytes32(bytes memory input) internal pure returns (bytes32 output) {
+    return bytes32(input);
+  }
+
+  function toUint256(bytes memory input) internal pure returns (uint256 output) {
+    return uint256(bytes32(input));
+  }
+
+  function toUint32(bytes memory input) internal pure returns (uint32 output) {
+    return uint32(bytes4(input));
+  }
+
+  function toAddress(bytes memory input) internal pure returns (address output) {
+    return address(bytes20(input));
+  }
+
+  function toBytes4(bytes memory input) internal pure returns (bytes4 output) {
+    return bytes4(input);
+  }
+
+  function toUint8(bytes memory input) internal pure returns (uint8 output) {
+    return uint8(input[0]);
+  }
+
+  /**
+   * Converts a `bytes` memory blob to a `bytes32` memory array.
+   */
+  function toBytes32Array(bytes memory input) internal pure returns (bytes32[] memory output) {
+    output = new bytes32[](Utils.divCeil(input.length, 32));
+    for (uint256 i = 0; i < output.length; i++) {
+      output[i] = toBytes32(input, i * 32);
+    }
+    return output;
+  }
+
+  /**
+   * Converts a `bytes` memory blob to a `SchemaType` memory array.
+   */
+  function toSchemaTypeArray(bytes memory input) internal pure returns (SchemaType[] memory output) {
+    output = new SchemaType[](input.length);
+    for (uint256 i = 0; i < output.length; i++) {
+      output[i] = SchemaType(uint8(input[i]));
+    }
+    return output;
+  }
+
+  /**
+   * Converts a tightly packed uint32 array into a regular uint32 array.
+   */
+  function toUint32Array(bytes memory input) internal pure returns (uint32[] memory output) {
+    return Cast.toUint32Array(Buffer_.fromBytes(input).toArray(4));
+  }
+
+  /**
+   * Converts a tightly packed address array into a regular address array.
+   */
+  function toAddressArray(bytes memory input) internal pure returns (address[] memory output) {
+    return Cast.toAddressArray(Buffer_.fromBytes(input).toArray(20));
+  }
+
+  /**
+   * Converts a tightly packed bytes24 array into a regular bytes24 array.
+   */
+  function toBytes24Array(bytes memory input) internal pure returns (bytes24[] memory output) {
+    return Cast.toBytes24Array(Buffer_.fromBytes(input).toArray(24, true));
+  }
+
+  /************************************************************************
+   *
+   *    UTILS
+   *
+   ************************************************************************/
+
+  function equals(bytes memory a, bytes memory b) internal pure returns (bool) {
+    if (a.length != b.length) {
+      return false;
+    }
+    return keccak256(a) == keccak256(b);
+  }
+
+  function setLengthInPlace(bytes memory input, uint256 length) internal pure returns (bytes memory) {
```

Comment:
```suggestion
  function setLengthInPlace(bytes memory input, uint256 length) internal pure returns (bytes memory) {
    assert(length <= input.length);
```
changes cost from 16 to 82

Reply:
This started out as a purely internal function, but it seems not to be used anywhere anymore. Will remove.